server/src/core/database-service.ts

```diff
@@ -207,126 +207,135 @@ export class DatabaseService {
 
   // Check if a transcript already exists for a specific videoId
   public async doesTranscriptExist(videoId: string): Promise<boolean> {
-    const idPattern = `youtube_${videoId}`;
+    const documentName = `youtube:${videoId}`;
     
     return this.fallbackService.withFallback(
       'doesTranscriptExist',
       // Fallback function
       () => {
-        // Search in-memory documents for the videoId in the id field
+        // Search in-memory documents for the videoId in the documentName field
         return Array.from(inMemoryDocuments.values())
           .some(item => {
             const doc = item.document;
-            return doc.id.startsWith(idPattern);
+            return doc.documentName === documentName;
           });
       },
       // Qdrant function
       async () => {
-        // Search for documents with an ID that starts with the pattern
+        // Search for documents with the documentName matching youtube:videoId
         const filter = {
           must: [
             {
-              key: 'id',
+              key: 'documentName',
```

Comment:
no, why you can make a very disruption and incompatibility breaking change like this?